include/zephyr/location.h

```diff
@@ -0,0 +1,402 @@
+/**
+ * @file location.h
+ *
+ * @brief Public location API.
+ *
+ */
+
+/*
+ * Copyright (c) 2022, Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_LOCATION_H_
+#define ZEPHYR_INCLUDE_LOCATION_H_
+
+/**
+ * @brief Location Interface
+ * @defgroup location_interface Location Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Location data
+ *
+ * The latitude and longitude are in the unit micro degree. 1 micro
+ * degree corresponds to approx. 11cm along the earths surface.
+ *
+ * The accuracy describes the radius within which the real
+ * position is with a 68.2% certainty, in millimeters.
+ */
+struct location_position {
+	/** Latitudal position in microdegrees (0 to +-180000000) */
+	int32_t latitude;
+	/** Longitudal position in microdegrees (0 to +-180000000) */
+	int32_t longitude;
+	/** The uptime when the position was determined */
+	int64_t uptime;
+	/** Accuracy radius in millimeters */
+	uint32_t accuracy;
+};
+
+/**
+ * @brief Bearing data
+ *
+ * The accuracy describes the uncertainty of the bearing in +-
+ * millidegrees with a 68.2% certainty.
+ */
+struct location_bearing {
+	/** Bearing angle relative to True North in millidegrees (0 to 360000) */
+	uint32_t bearing;
+	/** The uptime when the bearing was determined */
+	int64_t uptime;
+	/** Accuracy in millidegrees */
+	uint32_t accuracy;
+};
+
+/**
+ * @brief Speed data
+ *
+ * The accuracy describes the uncertainty of the speed in +-
+ * millimeters pr second with a 68.2% certainty.
+ */
+struct location_speed {
+	/** Speed over ground in millimeters pr second */
+	uint32_t speed;
+	/** The uptime when the speed was determined */
+	int64_t uptime;
+	/** Accuracy in millimeters pr second */
+	uint32_t accuracy;
+};
+
+/**
+ * @brief Altitude data
+ *
+ * The accuracy describes the uncertainty of the altitude
+ * in +- millimeters with a 68.2% certainty.
+ */
+struct location_altitude {
+	/** Altitude above mean sea level in millimeters */
+	int32_t altitude;
```

Comment:
Is MSL the best reference? Most GNSS receivers output altitude with respect to the WGS-84 ellipsoid AFAIK

Reply:
We should use the WGS84 reference ellipsoid, seems like a more stable option, i overlooked it in the RMC message, which also has it in MSL. I will change it.

Reply:
Updated